### ISSUE TITLE
[DOCS] customization_spec is missing from the VMware cloud docs

### ISSUE THREAD

**Issue by nicholasmhughes** [saltstack/salt#57933]
**Description**
Docs here:

https://docs.saltstack.com/en/master/topics/cloud/vmware.html

talks about enabling customization, but don't document the `customization_spec` parameter that points to how to customize the instance.

https://github.com/saltstack/salt/blob/master/salt/cloud/clouds/vmware.py#L2805

**Comment by ScriptAutomate** [comment]
The documentation generated for `salt.cloud.clouds.vmware` and the `create()` function:
- https://docs.saltstack.com/en/master/ref/clouds/all/salt.cloud.clouds.vmware.html#salt.cloud.clouds.vmware.create

Which only includes a link to the profile config here:
- https://docs.saltstack.com/en/master/topics/cloud/vmware.html#profiles

But `customization_spec` is ultimately called from here:
- https://github.com/saltstack/salt/blob/master/salt/utils/vmware.py#L354

It looks like it is using the following:
- https://pubs.vmware.com/vi3/sdk/ReferenceGuide/vim.CustomizationSpecManager.html

With more information about vmware custom specifications here:
- https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.vm_admin.doc/GUID-EB5F090E-723C-4470-B640-50B35D1EC016.html

This needs an SME, as I don't know what needs to be present in the `customization_spec` definition in the following locations:
- https://github.com/saltstack/salt/blob/master/doc/topics/cloud/vmware.rst
- Potentially elsewhere in docstrings in the other above links

**Comment by sagetherage** [comment]
@nicholasmhughes maybe the Cloud Working Group can help here and we can collaborate between the WGs - Docs and Cloud :)

**Comment by nicholasmhughes** [comment]
yeah... I mostly put this issue in to remind myself to eventually get around to it (unless someone steals it from me)

**Comment by sagetherage** [comment]
Great! We do have some tech writers on staff with SaltStack, but they are a small group and split between many teams, so great reason to collaborate here!

**Comment by amalaguti** [comment]
I ran into the same issue today and was testing customization_spec setting. 
A positive side effect of being able to select a customization spec, is that the instance will have its hostname set properly, matching the minion id (if your spec is set to use the virtual machine name for Computer name)
So no need to run a task (most customers use a state during start up) to set hostname on post deployment.
